frontend/src/metabase/modes/components/drill/PivotDrill/PivotDrill.tsx

```diff
@@ -0,0 +1,73 @@
+import React, { useState } from "react";
+import { t } from "ttag";
+import type {
+  ClickActionPopoverProps,
+  DrillOptions,
+  PopoverClickAction,
+} from "metabase/modes/types";
+
+import type { PivotByDrillOption } from "./types";
+import PivotByTimeDrill from "./PivotByTimeDrill";
+import PivotByLocationDrill from "./PivotByLocationDrill";
+import PivotByCategoryDrill from "./PivotByCategoryDrill";
+import PivotDrillPopover from "./PivotDrillPopover";
+import { ActionIcon, ClickActionButton } from "./PivotDrillPopover.styled";
+
+const PivotDrill = ({
+  question,
+  clicked,
+}: DrillOptions): PopoverClickAction[] => {
+  const categoryDrillOptions = PivotByCategoryDrill({ question, clicked });
```

Comment:
It seems that `PivotByCategoryDrill` etc are no longer "entry points" and the only real entry point is now `PivotDrill`. If that's the case, I believe we should not have such "components" at all move logic within them to `metabase-lib`. Ideally we would call a single `metabase-lib` method here instead of calling 3 drills